project/ShoelaceTranslator.scala

```diff
@@ -593,9 +593,11 @@ class ShoelaceTranslator(
       //  println(s"WARNING: scalaAttrInputType: Multi-element input not supported for attr `${attr.attrName}` in tag `${tagName}`.")
       //  "Element" // #nc or Element[], but how to express that...
     } else {
-      printableTypes.toSet match {
-        case s if s == Set(Def.JsCustomType("Date"), Def.JsStringType) => "js.Date" // I'm mapping from a JS Dom type to the Scala type. Here, I only want to work with `js.Date`s in Scala, not a union of `js.Date | String`
-        case _ => throw new Exception(s"ERROR: scalaPropInputTypeStr does not support multiple printable types in prop `${prop.propName}` in tag `${tagName}`: ${printableTypes.mkString(", ")}")
+      if (printableTypes.toSet == Set(Def.JsCustomType("Date"), Def.JsStringType)) {
+        // I'm mapping from a JS Dom type to the Scala type. Here, I only want to work with `js.Date`s in Scala, not a union of `js.Date | String`
+        "js.Date"
+      } else {
+        throw new Exception(s"ERROR: scalaPropInputTypeStr does not support multiple printable types in prop `${prop.propName}` in tag `${tagName}`: ${printableTypes.mkString(", ")}")
```

Comment:
Right now, this is a very Shoelace-specific test (this code is changed to return a union in 518b29c8eb689b073efbeb745b1e13915993b349). After working with the code generation for a little bit, I think I have some ideas for how we could make this generic across other component libraries, reduce duplication in this generator code, and more seamlessly handle JS unions and other special types. It might be early to propose something with that sweeping of a change now, but it's something I'll poke around with in the background while we get a feel for how to best help Nikita.